SimPEG/Utils/matutils.py

```diff
@@ -37,6 +40,9 @@ def mkvc(x, numDims=1):
 
 def sdiag(h):
     """Sparse diagonal matrix"""
+    if isinstance(h, Zero):
+        return h
```

Comment:
I guess it only matters if zero has a shape to it, which I don't think we have yet..?